mint-client/src/clients/gateway.rs

```diff
@@ -226,75 +229,27 @@ impl GatewayClient {
             .collect()
     }
 
-    // TODO: improve error propagation on tx transmission
     /// Waits for a outgoing contract claim transaction to be confirmed and retransmits it
     /// periodically if this does not happen.
-    ///
-    /// # Panics
-    /// * If the contract with the given ID isn't in the database. Only call this function with
-    ///   pending claimed contracts returned by `list_pending_claimed_outgoing`.
-    /// * If the task can not be completed in reasonable time meaning something went horribly
-    ///   wrong (should not happen with an honest federation).
-    pub async fn await_claimed_outgoing_accepted(&self, contract_id: ContractId) {
-        let transaction: Transaction = self
-            .context
+    pub async fn await_outgoing_contract_claimed(
+        &self,
+        contract_id: ContractId,
+        outpoint: OutPoint,
+    ) -> Result<()> {
+        self.context
+            .api
+            .await_output_outcome::<OutgoingContractOutcome>(outpoint, Duration::from_secs(10))
```

Comment:
Passing generic `OutgoingContractOutcome` here basically asserts that it was the type of output we were expecting